plugins/catalog-import/src/components/ImportStepper/defaults.tsx

```diff
@@ -169,74 +170,97 @@ export function defaultGenerateStepper(
-                      name="componentName"
-                      label="Name of the created component"
-                      placeholder="my-component"
-                      margin="normal"
-                      variant="outlined"
-                      fullWidth
-                      inputRef={register({ required: true })}
-                      error={Boolean(errors.componentName)}
-                      required
-                    />
+                      <Box marginTop={2}>
+                        <Typography variant="h6">
+                          Entity Configuration
+                        </Typography>
+                      </Box>
 
-                    <AutocompleteTextField
-                      name="owner"
-                      control={control}
-                      errors={errors}
-                      options={groups || []}
-                      loading={groupsLoading}
-                      loadingText="Loading groups…"
-                      helperText="Select an owner from the list or enter a reference to a Group or a User"
-                      errorHelperText="required value"
-                      textFieldProps={{
-                        label: 'Entity Owner',
-                        placeholder: 'Group:default/my-group',
-                      }}
-                      rules={{ required: true }}
-                      required
-                    />
-                  </>
-                )}
+                      <TextField
+                        name="componentName"
+                        label="Name of the created component"
+                        placeholder="my-component"
+                        margin="normal"
+                        variant="outlined"
+                        fullWidth
+                        inputRef={register({ required: true })}
+                        error={Boolean(errors.componentName)}
+                        required
+                      />
+
+                      {!watchUseCodeowners && (
+                        <AutocompleteTextField
+                          name="owner"
+                          control={control}
+                          errors={errors}
+                          options={groups || []}
+                          loading={groupsLoading}
+                          loadingText="Loading groups…"
+                          helperText="Select an owner from the list or enter a reference to a Group or a User"
+                          errorHelperText="required value"
+                          textFieldProps={{
+                            label: 'Entity Owner',
+                            placeholder: 'Group:default/my-group',
+                          }}
+                          rules={{ required: true }}
+                          required
+                        />
+                      )}
+
+                      <CheckboxField
+                        name="useCodeowners"
+                        inputRef={register}
```

Comment:
This should be register() which should also setup the binding to the field that is specified by the name prop. You then shouldn't need to have the onchange handler.

Reply:
Sure, if I removed the wrapper component then that would work inline here. 👍